### ISSUE TITLE
Armored Core 2 Another Age pnach not working (compatibility with Widescreen hack)

### ISSUE THREAD

**Issue by TheFoolZer0** [VanLaser/analog-controls-for-armored-core-ps2#5]
I tried to use the cheat to enable true analog but it doesn't work. I have the correct file in the same place and PCX2 recognizes it but it doesn't work.

**Comment by VanLaser** [comment]
I just tested the latest USA AC2AA .pnach on a different machine, and it works as intended. Since you provide zero details about your setup and experience with PCSX2, I'll just give some general tips (I can't magically "see" into your situation):

- restart the game after you enable/disable the cheat
- have a recent version of PCSX2 (the nightly version is recommended); you can even have a 2nd, portable PCSX2 setup for this nightly version for testing
- make sure you use a supported AC2AA ISO (for example, try the clean, USA version - CRC of the ISO should be: BD992F5C)
- check your controller settings in PCSX2, first to see if they are mapped at all to your buttons (if you are using PCSX2 for the first time, it's possible the controller is not set up at all at first run), or (if you played AC before) check if you don't have conflicting mappings that make for example the right analog output some other controller buttons instead of doing their default work
- make sure you actually enable the cheat for the game (at least the nightly PCSX2 version should show "1 cheat enabled" when you start the game)

If all fails, there's also a AC discord link on reddit armoredcore's channel, you can join that and come to the "oldgen" channel for more questions.

**Comment by TheFoolZer0** [comment]
Thanks, I figured it out its because I enabled wide screen patches and it worked for AC2 which made me assume that it would work for AC2 AA. I now did some investigation and it most likely worked because either the true analog patch was used before the wide screen patch which was able to be applied which let me play AC2 with 2 patches. This doesn't explain why this "exploit" didn't work for AC2: AA.

When I boot up AC2 with out using a save state analog controls don't work and when I boot up the save state it works fine. So yea this was weird.  Time to play the rest of the games that need this patch in 4:3. Excuse my ignorance.

**Comment by TheFoolZer0** [comment]
Wait actually I got two different ISOs and their crcs are supposed to be BD992F5C but my two copies from different sites say 9545216B

**Comment by VanLaser** [comment]
Glad you worked it out!

FWIW, at least with PCSX2 Nightly on this machine, AC2AA USA version, I was able to enable the Widescreen patch (the one that PCSX2 shows by default in the "patches" window) and at least the garage "AC Test" was working alright, with the dual analogs pnach enabled in the "Cheats" section.

You can also try using the PPF format to make yourself a patched ISO and then see if the widescreen hack works on that (it could alter the order of the patching operations, as you described above).

That said, it IS possible the widescreen patch interacts negatively with the dual analog patch. The fact that it worked for AC2 explains nothing - these kind of patches alter the game memory, so they can step over each other's toes so to speak, or not - and it's different for each executable. In other words, you can't assume if it works for one AC version, it will still work for the next one.

**Comment by TheFoolZer0** [comment]
When I was checking the CRCs when I deleted the game and tried a fresh ISO is doesn't match the CRC it is supposed to be which is really weird and I tried from 2 different sources too which explains why the pnach was not working. I think something could be wrong with my PCSX2 nightly or those 2 sources(Vimm's lair and myrient) have the wrong version which seems very unlikely.

Finally with the PPF patch PCSX2 doesn't recognize the game and doesn't apply any patches to it. I even tried to find the pnach files online and it juse says unlabeled patches and I cant even check then on or off. I'll try manually turning it on if it affects anything.

**Comment by VanLaser** [comment]
If you are using the CRCs shown by PCSX2, I think they are not the ISO CRCs, but the main executable CRC! So you can't use those to compare, you would rather need a separate CRC tool to use on the ISO file to compute its CRC value. But the names of the pnaches themselves also contain the CRC of the executables (not the ISOs), if they match what PCSX2 shows, they should be probably ok. Long story short, the pnaches won't apply to a wrong CRC exe (they would show as disabled). I mentioned the CRC before I knew what your problem is, I think they might have sidetracked you.

To use a pnach on the PPF-patched ISO, you have to change the name of the pnach to fit the CRC of the new executable (shown by PCSX2 in the "Game Properties" window).

**Comment by TheFoolZer0** [comment]
I quadrupled check the pnach matches the patched CRC shown and it just refuses to work. PCSX2 recognizes it and says the patch is active and when I start up the game, it isn't widescreen. I will try to find another patch maybe this one doesn't work.

**Comment by VanLaser** [comment]
Indeed, it could be either a wrong widescreen pnach, or - a possible bad interaction between the analogs mod and the widescreen pnach.

If it doesn't work and you like this kind of struggle, you may try the same process on the AC2AA European version, maybe the interaction with the widescreen hack is different (personally, I doubt it, but who knows).

**Comment by TheFoolZer0** [comment]
I tried the same process and it didn't work either so I can't tell what the problem is anymore I suppose it is just incompatibility with the wide screen patch.

**Comment by superblobster** [comment]
It's fully compatible with the widescreen patch. I just confirmed it works in 1.7.5717.
OP likely has an outdated emulator, messed up settings or a bad dump.
